### PR TITLE
Use orm.Context instead request context in background go routing

### LINKED ISSUE #12741: [proxy cache] sql error in the log
I see such error in the log:
```
Aug 11 12:54:47 172.18.0.1 core[4975]: 2020-08-11T12:54:47Z [ERROR] [/controller/proxy/controller.go:127]: failed to get manifest, error sql: transaction has already been committed or rolled back
```
Is it because the sequence of middlewares caused all DB calls in the proxy controller failed with such error?

### PR DESCRIPTION
**Pull request by stonezdj**
Fixes #12741: error sql: transaction has already been committed or rolled back

Signed-off-by: stonezdj <stonezdj@gmail.com>

### CODE REVIEW

File: src/controller/proxy/controller.go [hunk 1]
```diff
@@ -122,6 +123,7 @@ func (c *controller) ProxyManifest(ctx context.Context, p *models.Project, art l
 	}
 	// Push manifest in background
 	go func() {
+		ctx := orm.Context()
```
Comment (ywk253100) [inline src/controller/proxy/controller.go:126]: Use `func NewContext(ctx context.Context, o orm.Ormer) context.Context` instead? Or the other information in the context will be lost

Comment (heww) [reply src/controller/proxy/controller.go:126]: I don't think we need to use the `NewContext(ctx context.Context, o orm.Ormer)` here. What will happen when the HTTP goroutine was finished but this goroutine still not be scheduled by the runtime of Golang?

Comment (stonezdj) [reply src/controller/proxy/controller.go]: In current case, we need to get the user information from context, update to use the  NewContext(ctx context.Context, o orm.Ormer)

Comment (heww) [reply src/controller/proxy/controller.go]: I created an issue for the problem to get the operator of the event, https://github.com/goharbor/harbor/issues/12774.

If we don't get the operator in the `Resolve` of event metadata implicitly, I think we don't need the ctx from the HTTP request in this goroutine.

File: src/controller/proxy/controller.go [hunk 2]
```diff
@@ -122,6 +124,7 @@ func (c *controller) ProxyManifest(ctx context.Context, p *models.Project, art l
 	}
 	// Push manifest in background
 	go func() {
+		ctx := orm.NewContext(ctx, borm.NewOrm())
```
Comment (reasonerjt) [inline src/controller/proxy/controller.go:127]: The first parm should be context.Background?

File: src/controller/proxy/controller.go (hunk 1 above)
Comment (stonezdj) [reply src/controller/proxy/controller.go]: OK, I will add Operator field in PullArtifactEventMetadata

File: src/controller/proxy/controller.go (hunk 2 above)
Comment (stonezdj) [reply src/controller/proxy/controller.go]: updated

**Pull request merged**